### PR TITLE
DAOS-10367 tests: enable NVMe for VOS unit tests (#8822)

### PR DESCRIPTION
**Pull request by NiuYawei**
Enable NVMe when running VOS unit tests.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

**Review by wangshilong — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

File: src/vos/vos_common.c
Comment (ashleypittman) [inline src/vos/vos_common.c:687]: You need to use D_ASPRINTF()/D_FREE here, snprintf with a fixed size buffer isn't even remotely safe.

File: src/vos/vos_common.c
Comment (ashleypittman) [inline src/vos/vos_common.c:697]: access would be better than open for checking if the file exists.

**Review by ashleypittman — CHANGES_REQUESTED** [review]

**Comment by ashleypittman** [comment]
As per skype there are valgrind errors reported in Jenkins that need addressing as well.

**Comment by ashleypittman** [comment]
> As per skype there are valgrind errors reported in Jenkins that need addressing as well.

I take it back, master is broken.  I suggest you wait until we can resolve and then merge this with master.

**Comment by ashleypittman** [comment]
> > > As per skype there are valgrind errors reported in Jenkins that need addressing as well.
> > 
> > 
> > I take it back, master is broken. I suggest you wait until we can resolve and then merge this with master.
> 
> This is a clean cherry-pick for 2.2, so I suppose the broken stuff was in 2.2 as well?

No idea.  It could be a change to an upstream package as well.

File: src/vos/vos_common.c
Comment (jolivier23) [reply src/vos/vos_common.c:687]: These issues should be fixed on master branch first

Comment (jolivier23) [reply src/vos/vos_common.c:687]: There are two issues here.

One is that snprintf does not write null byte if the source string is longer.  In other places, we often "handle" this by always writing the last byte to 0.

But Ashley's proposal for using D_ASPRINTF is preferrable since it won't truncate the string at all.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
can you fix the things Ashley pointed out in master and here. I think this will introduce a coverity regression.

**Comment by NiuYawei** [comment]
> can you fix the things Ashley pointed out in master and here. I think this will introduce a coverity regression.

Sorry, missed that comment, I'll push a patch to master soon.

**Review by wangshilong — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**
